### ISSUE TITLE
Add possibility to clear single static file cache entries manually in boost mode via the backend module

### ISSUE THREAD

**Issue by kathangeorg** [lochmueller/staticfilecache#265]
## Feature Request

**Is your feature request related to a problem? Please describe.**
Using BoostMode in a project with multiple domains (trees) with hundrets of pages in each tree, clearing the cache (green or red) causes a queue with many many entries. The cronjob (e.g. as in #256) recreating the static file cache entries takes a very long time. 

Users tend to often clear the green or red cache instead of a single page cache. Then it´s problematic for our users not to see specific changes (e.g. after changes in an extension object like a product) on specific pages immediatly. Also one can say this is simply a user / workflow related problem, I think a technical solution would be beneficial.

**Describe the solution you'd like**
It would be helpfull to have the possibility to clear the static file cache entries for a single page (url) in the backend module of the extension.

**Describe alternatives you've considered**
Another solution could be to have the posibillity to rank important urls in the queue so they are regenerated first.

**Comment by j4k3** [comment]
Hey @lochmueller,

while I agree on everything you say regarding the technical aspects, @kathangeorg  still has a point here. 
Since my debugging session three weeks ago I have some things I wanted to point out, preferably in form of a pull request but I just can't find the time to implement and test this. So here's my two cents in form of a comment instead.

While it is correct that it would be TYPO3s task to figure out when a cache invalidates, TYPO3 is also unbelievably bad at it.
I've been trying to prevent my editors for pushing the clear cache buttons for years now, but with however finegrained you configure your system, who just won't stop pushing that big red button that says don't push me.

I set all my installations up in a way where 

    [backend.user.isLoggedIn]
        config.no_cache = 1
    [global]

is set in the main TypoScript so they always see the freshly generated page. There are, however, some edge cases where this doesn't work. I have, for example, some users that  post TYPO3 pages to social media platforms. Once The page is accessed anonymosly, staticfilecache writes the cached page and the editors then can't get changes on the page to show up in the social media post. When noticing that the page still doesn't update and they proceed to click on every cache flush button they can find. Repeatedly. This is one of the setups where there are almost 6000 cachable pages and flushing them all not only has a catastrophic impact on the performance of the whole page, but the single page in question they need updated is then lost somewhere in that queue and it may take hours for the staticfilecache worker to reach it. - This scenario may seem like an edge case, but it happens to me on a regular basis and I have yet to find any other solution, other than to look up the page in the file server and delete it there manually.

Long story short, I think the solution would be quite simple. The flush button for a specific page should not just push the page to the queue, but instead do so with the highest possible priority. That way it would at least be possible to get to rebuild a single page as quickly as possible.

I also think the editors (and admins) absolutely need to be informed about what he boostqueue runner is doing and to cancel that operation. I was trying to figure out if TYPO3 has a hook/signal for the cache context menu, but I couldn't find one so bar. But I think there should be a short informational message about the amount of pages currently in the boost queue and a button to clear the queue in case "someone" "accidentally" click the clear all pages button.

**Comment by lochmueller** [comment]
This line is missing: https://github.com/lochmueller/staticfilecache/blob/master/Classes/Cache/StaticFileBackend.php#L219 but I add a few ideas with  the last commit: https://github.com/lochmueller/staticfilecache/commit/d30f9c6d14e6ce2f1ca290cd6bde02cccea1f961 (also add a bit docuemntation).

The idea to get the page a high prio out of the page module is not possible directly. I would inntegrate it based on the cacheTags. If there is only a pageId_XXXX Tag, than the page get a higher prio in the queue.
